jobs/resend_invitation_email/worker.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+package resend_invitation_email
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v5/app"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/shared/mlog"
+)
+
+type ResendInvitationEmailWorker struct {
+	name    string
+	stop    chan bool
+	stopped chan bool
+	jobs    chan model.Job
+	App     *app.App
+}
+
+func (rse *ResendInvitationEmailJobInterfaceImpl) MakeWorker() model.Worker {
+	worker := ResendInvitationEmailWorker{
+		name:    ResendInvitationEmailJob,
+		stop:    make(chan bool, 1),
+		stopped: make(chan bool, 1),
+		jobs:    make(chan model.Job),
+		App:     rse.App,
+	}
+	return &worker
+}
+
+func (rseworker *ResendInvitationEmailWorker) Run() {
+	mlog.Debug("Worker started", mlog.String("worker", rseworker.name))
+
+	defer func() {
+		mlog.Debug("Worker finished", mlog.String("worker", rseworker.name))
+		rseworker.stopped <- true
+	}()
+
+	for {
+		select {
+		case <-rseworker.stop:
+			mlog.Debug("Worker received stop signal", mlog.String("worker", rseworker.name))
+			return
+		case job := <-rseworker.jobs:
+			mlog.Debug("Worker received a new candidate job.", mlog.String("worker", rseworker.name))
+			rseworker.DoJob(&job)
+		}
+	}
+}
+
+func (rseworker *ResendInvitationEmailWorker) Stop() {
+	mlog.Debug("Worker stopping", mlog.String("worker", rseworker.name))
+	rseworker.stop <- true
+	<-rseworker.stopped
+}
+
+func (rseworker *ResendInvitationEmailWorker) JobChannel() chan<- model.Job {
+	return rseworker.jobs
+}
+
+func (rseworker *ResendInvitationEmailWorker) cleanEmailData(emailStringData string) []string {
+	// emailStringData looks like this ["user1@gmail.com","user2@gmail.com"]
+	t := strings.Trim(strings.Trim(emailStringData, "["), "]")
+	u := strings.Split(t, ",")
+	var cleaned []string
+	for _, ss := range u {
+		cleaned = append(cleaned, ss[1:len(ss)-1])
+	}
+	return cleaned
+}
+
+func (rseworker *ResendInvitationEmailWorker) removeAlreadyJoined(teamID string, emailList []string) []string {
+	var notJoinedYet []string
+	for _, email := range emailList {
+		// check if the user with this email is on the system already
+		user, appErr := rseworker.App.GetUserByEmail(email)
+		if user == nil && appErr != nil {
```

Comment:
When the appErr is != nil the user should be nil always so no need to check that